### DISCUSSION TITLE
Is anybody else NOT using Home Assistant?

### DISCUSSION THREAD

**Discussion by Rujo** [amberelectric/public-api#163] [General]
I have some experience in using the programming language of Google Sheets which has the difficult name 'Google Apps Script' or GAS. For this reason I will avoid digging deep into Home Assistant but try an other and as I think more flexible approach.

With GAS I can call APIs like the Amber API and place the gathered data in a Google Sheet spreadsheet. 
The relevant Amber APIs are a bit difficult to use because they provide  essentially two datasets in one JSON array; the data for import and the data for export. But I am getting there.

At the moment I work on putting the data into the right place of the spreadsheet. Once this is done I can create charts.

The script will the also be able to switch individual devices on and off, if they have an API or I can control appliances with remote controlled power points. Being a long time Arduino fan, I can also hack a remote control to turn air conditioners on and off. That's at least the plan.

I am posting this here because I am interested to get your feedback. And maybe someone is willing to work with me on this development.

**Comment by rob0101** [comment]
Odd question, but I'm not using Home Assistant.

I have my own scripts to call the API's of Amber, Tesla (powerwall) and myEnergi (zappi) plus Apple Homekit via Apple Shortcuts.

I use Perl, since that's what I've used for 30 years.

Price based time shifting is how I make Amber work for me. When Amber's API expands I'll also like to manage when the powerwall charges, rests and discharges.

**Comment by freman** [comment]
I have been using home assistant but I'm about to migrate to some custom tooling, home assistant seems to update it's data at whenever it feels like it intervals which means finding out about a price spike 4 minutes before the pricing period ends lol.

**Reply by purcell-lab** [reply]
Home Assistant the first to alert me about tonight's price spike by pinging my phone after it had turned off all my heavy loads. I didn't receive an alert from the Amber app.

**Reply by freman** [reply]
My HA told me at 19:41 about the spike as it started exporting.

**Reply by rob0101** [reply]
Spikes can start late in the half hour giving no tool enough time to react.

**Reply by rob0101** [reply]
I have custom alerts based on actual 5 minute prices that are unusually high or low. Amber might predict something before I notice, but it may also predict something that fails to eventuate, so I prefer to respond to confirmed prices.

**Reply by herbi3** [reply]
I use python, Emoncms, influxdb and grafana.

Python, to pull the data from amber and all of my inverters and batteries, and used to control the inverters and batteries too.

Emoncms for number crunching and allowing the data to be accessed via api for other scripts (so amber doesn't get smashed, Emoncms caches the data)

Influxdb and grafana, store the data and present it on a dashboard

In my main python code, I have it look ahead by four hours for the general and solar pricing. So, if the current FiT price is above 27c and the fit price for exporting now will be higher than the average general import price for at least the next two hours, then it sends the signal to my batteries to dispatch.

This prevents me from exporting when I'll just end up buying back at a higher price once the batteries are depleted. I set two hours, because our 30kWh storage at maximum dispatch will be depleted after two hours.

It responds instantly, but downside is there's no forecasting for charging, so it does its best to charge during the day, then force charges between 3pm and 4pm so it will have at least 50% charge before 4pm

**Comment by Rujo** [comment]
As promised, here is my result using Google Sheets and Google Apps Script. https://docs.google.com/spreadsheets/d/1nKnPSQWKG7rnA-QKsfT2ZFtZlgX6IhWFjITb8dxo1K4/edit#gid=440828275
I have tried to show the price, quantity, and cost on the same timeline, both for import and export. As this would be too crowded in one chart, I chose to place the charts in close proximity and modify the timeline consistently.
I did the process in two steps to protect my privacy. One Google sheet reads the data from the API every 30 minutes and collects it in one spreadsheet. The second spreadsheet, linked above, reads the data from the first spreadsheet and presents them in a meaningful way.

I am planning to add a second page that deals with real-time data and price spike warnings.
![image](https://github.com/amberelectric/public-api/assets/8346461/fe6ccbcf-3df2-4e4c-b017-8989eb46cc21)

**Reply by Rujo** [reply]
For privacy, this chart is only a demo and not fully functional.

**Comment by petebitt99** [comment]
I'm using Athom's Homey Pro

**Comment by craigcurtin-dev** [comment]
I am using Node Red

**Comment by psmedley** [comment]
I'm using OpenHAB.  the 4.2.0 release will include a binding for Amber.

**Comment by Tim-nodo** [comment]
Not home assistant. I am using scripts I have written to control many things in my home and also work with smartshift to get good co-operation with EV charging, hot water heating etc. My scripts started off managing my battery for time of use about 6 years ago but have grown and adapted.
